tests/cli/test_workflow.py

```diff
@@ -785,3 +786,75 @@ def test_workflow_compose_execute(runner, project, run_shell, client):
     assert output[1] is None
 
     assert "xyz\n" == Path("output4").read_text()
+
+
+@pytest.mark.parametrize("provider", available_workflow_providers())
+@pytest.mark.parametrize(
+    "workflow, parameters, num_iterations",
+    [
+        ('echo "a" > output', {}, 0),
+        ('echo "a" > output', {"parameter-1": ["b", "c", "d"], "output-2": "output_{iter_index}"}, 3),
+        (
+            "head -n 1 Dockerfile > output",
+            {
+                "input-2": ["environment.yml", "Dockerfile", "requirements.txt"],
+                "n-1": ["3", "4"],
+                "output-3": "output_{iter_index}",
+            },
+            6,
+        ),
+        (
+            "head -n 1 Dockerfile > output",
+            {
+                "input-2@tag1": ["environment.yml", "Dockerfile", "requirements.txt"],
```

Comment:
I think one last case we need to test is mixing different tags, like `tag1` and `tag2` in the same command.